c10/core/SymFloat.cpp

```diff
@@ -88,4 +90,12 @@ double SymFloat::guard_float(const char* file, int64_t line) const {
   return a->guard_float(file, line);
 }
 
+SymInt SymFloat::floor() const {
+  if (!is_symbolic()) {
+    return int(data_);
```

Comment:
use floor, not int; when negative floor goes to negative infinity

Reply:
Changed, forgot that they represent more than just sizes